File: back-end/src/models/OrderItem.js
import DataTypes from 'sequelize';
import {
    sequelize
} from '../config/database';
import {
    Product
} from './Product';
import {
    Order
} from './Order';

const OrderItem = sequelize.define('order_items', {
    order_id: {
        type: DataTypes.INTEGER,
        allowNull: false
    },
    product_id: {
        type: DataTypes.STRING,
        allowNull: false
    },
    quantity: {
        type: DataTypes.STRING,
        allowNull: false
    }
}, {
    timestamps: false,
});

Product.hasMany(OrderItem, {
    foreignKey: 'product_id',
    sourceKey: 'id'
});
OrderItem.belongsTo(Product, {
    foreignKey: 'product_id',
    targetKey: 'id'
});

Order.hasMany(OrderItem, {
    foreignKey: 'order_id',
    sourceKey: 'id'
});
OrderItem.belongsTo(Order, {
    foreignKey: 'order_id',
    targetKey: 'id'
});

export {
    OrderItem
};

File: front-end/order_app/lib/models/order_item.dart
class OderItem {
  int _orderId, _productId, _quantity;

  int get orderId => _orderId;
  int get productId => _productId;
  int get quantity => _quantity;

  set orderId(int orderId) {
    this._orderId = orderId;
  }

  set productId(int productId) {
    this._productId = productId;
  }

  set quantity(int quantity) {
    this._quantity = quantity;
  }
}
